runtime/onert/core/src/ir/train/UseDefGenerator.test.cc

```diff
@@ -0,0 +1,2027 @@
+    std::vector<train::TrainingOperationIndex> expected_forwarding_fc_out_def{forwarding_fc_index};
+    const auto expected_forwarding_fc_out =
+      createUseDefChain(tgraph.operands().at(fc_out), expected_forwarding_fc_out_uses,
+                        expected_forwarding_fc_out_def);
+
+    std::vector<train::TrainingOperationIndex> expected_forwarding_ea_out_uses{
+      forwarding_ba_index, backwarding_ea_index, backwarding_ba_index};
+    std::vector<train::TrainingOperationIndex> expected_forwarding_ea_out_def{forwarding_ea_index};
+    const auto expected_forwarding_ea_out =
+      createUseDefChain(tgraph.operands().at(ea_out), expected_forwarding_ea_out_uses,
+                        expected_forwarding_ea_out_def);
+
+    std::vector<train::TrainingOperationIndex> expected_forwarding_y_pred_uses{
+      forwarding_loss_index, backwarding_loss_index};
+    std::vector<train::TrainingOperationIndex> expected_forwarding_y_pred_def{forwarding_ba_index};
+    const auto expected_forwarding_y_pred =
+      createUseDefChain(tgraph.operands().at(y_pred), expected_forwarding_y_pred_uses,
+                        expected_forwarding_y_pred_def);
+
+    std::vector<train::TrainingOperationIndex> expected_forwarding_y_true_uses{
+      forwarding_loss_index, backwarding_loss_index};
+    std::vector<train::TrainingOperationIndex> expected_forwarding_y_true_def{};
+    const auto expected_forwarding_y_true =
+      createUseDefChain(tgraph.operands().at(y_true), expected_forwarding_y_true_uses,
+                        expected_forwarding_y_true_def);
+
+    std::vector<train::TrainingOperationIndex> expected_forwarding_output_uses{};
+    std::vector<train::TrainingOperationIndex> expected_forwarding_output_def{
+      forwarding_loss_index};
+    const auto expected_forwarding_output =
+      createUseDefChain(tgraph.operands().at(output), expected_forwarding_output_uses,
+                        expected_forwarding_output_def);
+
+    std::vector<train::TrainingOperationIndex> expected_backwarding_input_uses{};
+    std::vector<train::TrainingOperationIndex> expected_backwarding_input_def{backwarding_fc_index};
+    const auto expected_backwarding_input = createUseDefChain(
+      tgraph.operands().at(input), expected_backwarding_input_uses, expected_backwarding_input_def);
+
+    std::vector<train::TrainingOperationIndex> expected_backwarding_weights_uses{};
+    std::vector<train::TrainingOperationIndex> expected_backwarding_weights_def{
+      backwarding_fc_index};
+    const auto expected_backwarding_weights =
+      createUseDefChain(tgraph.operands().at(weights), expected_backwarding_weights_uses,
+                        expected_backwarding_weights_def);
+
+    std::vector<train::TrainingOperationIndex> expected_backwarding_fc_out_uses{
+      backwarding_fc_index};
+    std::vector<train::TrainingOperationIndex> expected_backwarding_fc_out_def{
+      backwarding_ea_index, backwarding_ba_index};
```

Comment:
A back-prop tensor corresponding to a forwarding tensor that used in multiple nodes has multiple defs. The back-prop tensor would be accumulated in backwarding, but current trainable graph in onert cannot represent accumulating back-prop tensors. Allowing multiple defs is a workaround.